### PR TITLE
Add create and delete functions for composite api endpoints

### PR DESCRIPTION
**Pull request by jeremy-hanna**
This PR extends the work done on https://github.com/chulkilee/ex_force/pull/40 to support additional CRUD operations:
- [creates](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_create.htm)
- [deletes](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_delete.htm)

There are some minor changes to the docs around some unintuitive things in the Salesforce API. Namely their error codes, multiple type updates, and partial success states.

### CODE REVIEW

File: lib/ex_force.ex [hunk 1]
```diff
@@ -271,6 +303,32 @@ defmodule ExForce do
...
+
+  While the response code from Salesforce is often a 200, there may be failures mixed with
+  success records so use your own filter and retry logic for error_handling.
+
+  If more than 200 records need to be updated at once, try using the Bulk API:
+  see [Delete Multiple Records with Fewer Round-Trips](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_delete.htm)
+  """
+  @spec delete_sobjects(client, records :: list(sobject), all_or_none :: boolean) ::
+          {:ok, any} | {:error, any}
+  def delete_sobjects(client, ids, all_or_none \\ false)
```
Comment (jeremy-hanna) [inline lib/ex_force.ex:320]: I was thinking of pattern matching on the empty set for ids and returning an error which is why I pulled the function signature with the default up

I am not sure what the convention here is or if there is a need to however

File: test/ex_force_test.exs [hunk 2]
```diff
@@ -573,6 +573,193 @@ defmodule ExForceTest do
              {:error, :econnrefused}
   end
 
+  test "create_sobjects/3 - success", %{bypass: bypass, client: client} do
+    Bypass.expect_once(bypass, "POST", "/services/data/v40.0/composite/sobjects", fn conn ->
```
Comment (jeremy-hanna) [inline test/ex_force_test.exs:577]: these tests are a bit misleading - I believe the composite API is available post v42.0 (at least according to the historical [salesforce docs](https://developer.salesforce.com/docs/atlas.en-us.212.0.api_rest.meta/api_rest/resources_composite_sobjects_collections_delete.htm))

File: test/ex_force_test.exs [hunk 3]
```diff
@@ -613,6 +800,155 @@ defmodule ExForceTest do
...
+                  "success" => false,
+                  "errors" => [
+                    %{
+                      "statusCode" => "MALFORMED_ID",
+                      "message" => "Contact ID: id value of incorrect type: foo",
+                      "fields" => ["Id"]
+                    }
+                  ]
+                }
+              ]}
```
Comment (jeremy-hanna) [inline test/ex_force_test.exs:949]: the scenario where:
- the API request was well-formed
- the request was executed
- the request failed to a data condition
- every operation was rolled back

leads to a 200 from Salesforce and every record operation `success: false`

this is unintuitive could conceivably be returned as an `{:error, ...}` but I figured `{:ok, ...}` better reflected how Salesforce responded

**Comment by chulkilee** [comment]
I'll take a look soon - sorry for the deley! @jeremy-hanna

**Pull request closed without merging**